Fetch&rebase support for pull requests

kohsuke opened the pull request:
I work on projects on GitHub that receives a fair number of pull requests, and most of the time I need to make some slight adjustments on them before merging changes. I also do fetch+rebase when merging pull requests in an attempt to create more linear history. I find it easier to comprehend what was going on that way.

To that end, I added a command to do this in one shot:
- you run "github fetch-pull 3" where 3 indicates a pull request ID.
- the command looks at your remote and figure out which repository this is.
- it then fetches and rebases the pull request, then finishes.

If you find this command useful, would you please consider merging it?

kohsuke committed:
Added a command to fetch&rebase a pull request.

kohsuke committed:
Ignore rubymine files.

kohsuke committed:
improved error handling

kohsuke committed:
I noticed that git show-ref HEAD resolves to origin/HEAD, which isn't
what I'm expecting.

nixon committed:
add private repo support to fetch-pull

nixon commented:
Here's a commit that adds support for private repos to kohsuke's fetch-pull feature:

  https://github.com/nixon/github-gem/commit/d856a17f5c0c75c0e9f785ee501616b597cdfd1d

kohsuke committed:
Merge pull request #1 from nixon/fetch-pull_orgs

Support for private repositories.

kohsuke commented:
The change mentioned in the comment from nixon is now merged into this pull request.

nixon committed:
provide a fetch-pull merge alternative

add merge, rebase and do-nothing options to the fetch-pull command.  i
like being able to fetch a pull request by number, but sometimes i'd
rather not have the rebase happen automatically.

kohsuke committed:
Merge pull request #2 from nixon/fetch-pull_with_options

Fetch pull with options

kohsuke committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	.gitignore

kohsuke committed:
Switching to OAuth token

There's a lot more work to be done to convert other calls, but this is a
start.

kohsuke committed:
If the user is in org, check out via SSH.

kohsuke committed:
canonicalize the path so that link to bin can be a symlink

kohsuke committed:
improved the definition

kohsuke committed:
far simpler implementation of fetch-pull
